From d3934cfa45504183b80bb63a29f25d5d017b1d36 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 20 Nov 2019 11:51:55 +0100
Subject: [PATCH] PONRTSYS-5378: Deactivate RX on LAN ports by default

Deactivate the RX when the LAN ports are registered to DP, this is
activated when we call ndo_open on the interface.
This way all packets which are received while the interface is down are dropped.
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index bac61736e235..af4f36c42732 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -953,6 +953,9 @@ static int ltq_eth_init(struct net_device *dev)
 			dev->ethtool_ops = &serdes_ethtool_ops;
 	}
 
+	data.flag_ops |= DP_SUBIF_RX_FLAG;
+	data.rx_en_flag = netif_running(dev);
+
 	if (priv->lct_en == 1)
 		data.flag_ops |= DP_F_DATA_LCT_SUBIF;
 
